storage/flux/table_test.go

```diff
@@ -1 +1,3543 @@
+	if err := metaClient.Open(); err != nil {
+		close()
+		tb.Fatalf("failed to open meta client: %s", err)
+	}
+	closers = append(closers, metaClient)
+
+	// Create the database.
+	dbName := "db"
+	rpName := "rp"
+
+	// Run the setup function to create the series generator.
+	sg, tr := setupFn(dbName, rpName)
+	rp := &meta.RetentionPolicySpec{
+		Name:               rpName,
+		ShardGroupDuration: 24 * 7 * time.Hour * 52,
+	}
+	if _, err := metaClient.CreateDatabaseWithRetentionPolicy(dbName, rp); err != nil {
+		close()
+		tb.Fatalf("failed to create database: %s", err)
+	}
+
+	// Create the shard group for the data. There should only be one and
+	// it should include the entire time range.
+	sgi, err := metaClient.CreateShardGroup(dbName, rp.Name, tr.Start)
+	if err != nil {
+		close()
+		tb.Fatalf("failed to create shard group: %s", err)
+	} else if sgi.StartTime.After(tr.Start) || sgi.EndTime.Before(tr.End) {
+		close()
+		tb.Fatal("shard data range exceeded the shard group range; please use a range for data that is within the same year")
+	}
+
+	// Create a TSDB for the reader to access.
+	tsdbStore := tsdb.NewStore(rootDir)
+	if err := tsdbStore.Open(); err != nil {
+		close()
+		tb.Fatalf("failed to open TSDB store: %s", err)
+	}
+	closers = append(closers, tsdbStore)
+
+	// Write data into the shard.
+	if err := tsdbStore.CreateShard(dbName, rpName, sgi.Shards[0].ID, true); err != nil {
+		close()
+		tb.Fatalf("failed to create shard: %s", err)
+	}
+	shard := tsdbStore.Shard(sgi.Shards[0].ID)
+	sfile, err := shard.SeriesFile()
+	if err != nil {
+		close()
+		tb.Fatalf("failed to get series file: %s", err)
+	}
+	sfile.DisableCompactions()
+	shardPath := shard.Path()
+	if err := os.MkdirAll(shardPath,0700); err != nil {
+		close()
+		tb.Fatalf("failed to create shard directory: %s", err)
+	}
+
+	if err := writeShard(sfile, sg, sgi.Shards[0].ID, shardPath); err != nil {
+		close()
+		tb.Fatalf("failed to write shard: %s", err)
+	}
+
+	// Run the partition compactor on the series file.
+	for i, p := range sfile.Partitions() {
+		c := tsdb.NewSeriesPartitionCompactor()
+		if err := c.Compact(p); err != nil {
+			close()
+			tb.Fatalf("failed to compact series file %d: %s", i, err)
+		}
+	}
+
+	// Reset the TSDB store.
+	if err := tsdbStore.Close(); err != nil {
+		close()
+		tb.Fatalf("failed to close TSDB store: %s", err)
+	}
+	if err := tsdbStore.Open(); err != nil {
+		close()
+		tb.Fatalf("failed to reopen TSDB store: %s", err)
+	}
+
+	store := storage.NewStore(tsdbStore, metaClient)
+	reader := storageflux.NewReader(store)
+	return &StorageReader{
+		Database:        dbName,
+		RetentionPolicy: rpName,
+		Bounds: execute.Bounds{
+			Start: values.ConvertTime(tr.Start),
+			Stop:  values.ConvertTime(tr.End),
+		},
+		Close:  close,
+		Reader: reader,
+	}
+}
+
+func (r *StorageReader) ReadWindowAggregate(ctx context.Context, spec influxdb.ReadWindowAggregateSpec, alloc *memory.Allocator) (influxdb.TableIterator, error) {
```

Comment:
From here down, the only differences from the source 2.x commit are:
1. Naming of some things from import/interface differences
2. Using `Database` and `RetentionPolicy` instead of `OrganizationID` and `BucketID`